### PR TITLE
Move docker files to docker folder

### PR DESCRIPTION
**Pull request by obrask1**
- [x] Move docker files from `src/fleetlearning` to `docker` 
- [x] merge the old `docker/readme` into the old `src/fleetlearning/readme` and move this merged readme into `docker/readme`

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# Deploy with Docker
+## Setting up docker containers manually
+
+Ensure the correct ip addresses are specified:
+- Make sure the `static_params.py` file is updated with the available AGX IP (e.g. localhost if everything is deployed in the same node) and number of clients
```
Comment (jonasandersson10) [inline docker/README.md:5]:
```suggestion
- Make sure the `static_params.py` file is updated with the available AGX IPs (e.g. 127.0.0.1 if everything is deployed in the same node) and number of clients
```

File: docker/README.md [hunk 2]
```diff
@@ -16,7 +22,90 @@ Send the `.tar` file to the other host and load it by
...
 
+Finally, run the container
 
 ```docker
 docker run -v /mnt/ZOD2:/mnt/ZOD2 <image name> --cpus=<num_cpus> -m <num_of_GB>GB
-```
\ No newline at end of file
+```
+
+Then execute the main scripts
```
Comment (jonasandersson10) [inline docker/README.md:31]:
```suggestion
Then execute the main scripts inside corresponding container:
```

Comment (jonasandersson10) [reply docker/README.md:31]: or maybe you're referring to one-container deployment? might want to clarify either case

File: docker/README.md [hunk 3]
```diff
@@ -16,7 +22,90 @@ Send the `.tar` file to the other host and load it by
...
+To connect the other worker nodes, copy the token printed out in the master node terminal. On each worker node host run:
+
+```docker
+docker swarm join --token <ref-token-from-master-node> 
+```
+
+In a multiple hosts deployment, ensure the correct ip addresses are specified:
+- Make sure the `static_params.py` file is updated with the available AGX IPs and number of clients
+- Run the script `src/fleetlearning/configure_node_ips.py`. A YAML file will be created, containing the mapping between clients and AGXs. Furthermore, the docker-compose file will be automatically updated.
+- In `client_main.py`, change the `server_ip` to the corresponding IP addresses of the server node
```
Comment (jonasandersson10) [inline docker/README.md:52]:
```suggestion
- In `client_main.py`, change the `server_ip` and `agx_ip` to the corresponding IP addresses of the server and scheduler node, respectively.
```
right?

File: docker/README.md [hunk 4]
```diff
@@ -16,7 +22,90 @@ Send the `.tar` file to the other host and load it by
...
+- In `server_main.py`, set the server ip to `server_ip="0.0.0.0"`
+- In `scheduler_main.py`, set the agx ip to `agx_ip="0.0.0.0"`
+
+In a multiple hosts deployment, a pre-built image needs to be accessible on each worker node. Hence, start with building the right image on each worker by
```
Comment (jonasandersson10) [inline docker/README.md:56]: Don't know if we should make another PR for this, but here we should probably add another header with description on how to pull from our image registry on docker hub. We only need to add `docker login -u <account_name> ` and then `docker pull <account_name>/<image>:<tag>`. Then we should probably also change the default image in the compose files to pull directly from the hub.

File: docker/README.md (hunk 3 above)
Comment (obrask1) [reply docker/README.md:52]: Since my last PR, the server sends the AGX IP as a message to the client. So that is not needed anymore. Instead, its the self.DEVICE_DICT in static_params that determines what AGX ips that should be used.

File: docker/README.md (hunk 4 above)
Comment (obrask1) [reply docker/README.md:56]: Yeah agree, but maybe a separate PR for extended docker hub support? My intention here was only to refactor and merge the current readmes.

File: docker/README.md (hunk 3 above)
Comment (jonasandersson10) [reply docker/README.md:52]: ah yes, I forgot about that. 👍 sorry!
